Sources/TuistKit/Services/TargetImportsScanner.swift

```diff
@@ -28,7 +33,7 @@ final class TargetImportsScanner: TargetImportsScanning {
             filesToScan.append(contentsOf: headers.project)
         }
         var imports = Set(
-            try await filesToScan.concurrentMap { file in
+            try await filesToScan.concurrentMap(maxConcurrentTasks: Self.maxConcurrentTasks) { file in
```

Comment:
```suggestion
            try await filesToScan.concurrentMap(maxConcurrentTasks: 100) { file in
```

No need to limit ourselves to the actual number of cores – Swift concurrency ensures that the number of threads are handled efficiently. 100 is the same arbitrary limit we've chosen for hashing files, so we're pretty sure we wouldn't exhaust the number of opened files with this number.

Reply:
Good point! I can not apply magic number so I made additional constant to improve code readability + applied it into the place you mentioned

Reply:
also I wanted to notice using NIOCore.System.coreCount is not about limiting ourselves - I suppose at some point macos could manage more cores than 100 =)

Reply:
There's imho no benefit to using a constant here. I'd lean to applying the suggestion I proposed and call it. Either way, the `maxConcurrentTasks` should go away in the long term.

Reply:
I would preffer to keep this PR as is. Feel free to close it and make the bugfix with your own way - if disagree.